[BUG] test_docker_distribution Integration Test Times Out Due to Cluster Start Failure

The `test_docker_distribution` integration test fails as the Docker container does not come up healthy:

```
it/distribution_test.py::test_docker_distribution[os-it]
   ____                  _____                      __       ____                  __                         __
  / __ \____  ___  ____ / ___/___  ____ ___________/ /_     / __ )___  ____  _____/ /_  ____ ___  ____ ______/ /__
 / / / / __ \/ _ \/ __ \\__ \/ _ \/ __ `/ ___/ ___/ __ \   / __  / _ \/ __ \/ ___/ __ \/ __ `__ \/ __ `/ ___/ //_/
/ /_/ / /_/ /  __/ / / /__/ /  __/ /_/ / /  / /__/ / / /  / /_/ /  __/ / / / /__/ / / / / / / / / /_/ / /  / ,<
\____/ .___/\___/_/ /_/____/\___/\__,_/_/   \___/_/ /_/  /_____/\___/_/ /_/\___/_/ /_/_/ /_/ /_/\__,_/_/  /_/|_|
    /_/

[INFO] Preparing for test execution ...
[ERROR] Cannot execute_test. No healthy running container after 600 seconds!
	Traceback (most recent call last):
  File "/Users/engechas/OpenSearch-Benchmark/.tox/py38/lib/python3.8/site-packages/osbenchmark/builder/builder.py", line 563, in receiveMsg_StartNodes
    self.builder.start_engine()
  File "/Users/engechas/OpenSearch-Benchmark/.tox/py38/lib/python3.8/site-packages/osbenchmark/builder/builder.py", line 682, in start_engine
    self.nodes = self.launcher.start(self.node_configs)
  File "/Users/engechas/OpenSearch-Benchmark/.tox/py38/lib/python3.8/site-packages/osbenchmark/builder/launcher.py", line 53, in start
    self._start_process(binary_path)
  File "/Users/engechas/OpenSearch-Benchmark/.tox/py38/lib/python3.8/site-packages/osbenchmark/builder/launcher.py", line 73, in _start_process
    self._wait_for_healthy_running_container(container_id, DockerLauncher.PROCESS_WAIT_TIMEOUT_SECONDS)
  File "/Users/engechas/OpenSearch-Benchmark/.tox/py38/lib/python3.8/site-packages/osbenchmark/builder/launcher.py", line 93, in _wait_for_healthy_running_container
    raise exceptions.LaunchError(msg)
osbenchmark.exceptions.LaunchError: No healthy running container after 600 seconds!


Getting further help:
*********************
* Check the log files in /Users/engechas/.benchmark/logs for errors.
* Read the documentation at https://opensearch.org/docs.
* Ask a question on the forum at https://discuss.opendistrocommunity.dev/.
* Raise an issue at https://github.com/opensearch-project/OpenSearch-Benchmark/issues and include the log files in /Users/engechas/.benchmark/logs.

---------------------------------
[INFO] FAILURE (took 668 seconds)
---------------------------------
FAILED
```

The container's OpenSearch log shows:
```
[2022-02-23T16:29:26,314][ERROR][o.o.b.OpenSearchUncaughtExceptionHandler] [opensearch-node1] uncaught exception in thread [main]
org.opensearch.bootstrap.StartupException: java.lang.IllegalArgumentException: setting [cluster.initial_master_nodes] is not allowed when [discovery.type] is set to [single-node]
uncaught exception in thread [main]
at org.opensearch.bootstrap.OpenSearch.init(OpenSearch.java:186) ~[opensearch-1.0.0.jar:1.0.0]
at org.opensearch.bootstrap.OpenSearch.execute(OpenSearch.java:173) ~[opensearch-1.0.0.jar:1.0.0]
at org.opensearch.cli.EnvironmentAwareCommand.execute(EnvironmentAwareCommand.java:99) ~[opensearch-1.0.0.jar:1.0.0]
at org.opensearch.cli.Command.mainWithoutErrorHandling(Command.java:140) ~[opensearch-cli-1.0.0.jar:1.0.0]
at org.opensearch.cli.Command.main(Command.java:103) ~[opensearch-cli-1.0.0.jar:1.0.0]
at org.opensearch.bootstrap.OpenSearch.main(OpenSearch.java:139) ~[opensearch-1.0.0.jar:1.0.0]
at org.opensearch.bootstrap.OpenSearch.main(OpenSearch.java:105) ~[opensearch-1.0.0.jar:1.0.0]
```

This behavior is observed on both Linux and Mac

--- a/osbenchmark/resources/docker-compose.yml.j2
+++ b/osbenchmark/resources/docker-compose.yml.j2
@@ -15,7 +15,6 @@ services:
       - cluster.name=opensearch-cluster
       - node.name=opensearch-node1
       - discovery.seed_hosts=opensearch-node1
-      - cluster.initial_master_nodes=opensearch-node1
       - DISABLE_INSTALL_DEMO_CONFIG=true
       - bootstrap.memory_lock=true
       - "OPENSEARCH_JAVA_OPTS=-Xms512m -Xmx512m"
@@ -39,6 +38,11 @@ services:
       - 9600:9600
     networks:
       - opensearch-net
+    healthcheck:
+          test: curl -f http://localhost:{{http_port}} -u admin:admin --insecure
+          interval: 5s
+          timeout: 2s
+          retries: 10
 
 volumes:
   opensearch-data1:

--- a/tests/builder/provisioner_test.py
+++ b/tests/builder/provisioner_test.py
@@ -436,7 +436,6 @@ services:
       - cluster.name=opensearch-cluster
       - node.name=opensearch-node1
       - discovery.seed_hosts=opensearch-node1
-      - cluster.initial_master_nodes=opensearch-node1
       - DISABLE_INSTALL_DEMO_CONFIG=true
       - bootstrap.memory_lock=true
       - "OPENSEARCH_JAVA_OPTS=-Xms512m -Xmx512m"
@@ -457,6 +456,11 @@ services:
       - 9600:9600
     networks:
       - opensearch-net
+    healthcheck:
+          test: curl -f http://localhost:39200 -u admin:admin --insecure
+          interval: 5s
+          timeout: 2s
+          retries: 10
 
 volumes:
   opensearch-data1:
@@ -503,7 +507,6 @@ services:
       - cluster.name=opensearch-cluster
       - node.name=opensearch-node1
       - discovery.seed_hosts=opensearch-node1
-      - cluster.initial_master_nodes=opensearch-node1
       - DISABLE_INSTALL_DEMO_CONFIG=true
       - bootstrap.memory_lock=true
       - "OPENSEARCH_JAVA_OPTS=-Xms512m -Xmx512m"
@@ -524,6 +527,11 @@ services:
       - 9600:9600
     networks:
       - opensearch-net
+    healthcheck:
+          test: curl -f http://localhost:39200 -u admin:admin --insecure
+          interval: 5s
+          timeout: 2s
+          retries: 10
 
 volumes:
   opensearch-data1:

